### PR TITLE
Animation modernization with built in easing

### PR DESCRIPTION
**Pull request by rwaldron**
@dtex As I start to think about modernizing the hardware classes themselves, I've decided to do a "practice" round on Animation, because it's one of J5 components I'm least familiar with. Please review this rigorously!!

### CODE REVIEW

File: lib/easing.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Object.keys(ease).forEach(key => {
+  exportables[key.toLowerCase()] = ease[key];
+});
+
+module.exports = new Proxy({}, {
+  get(target, property) {
+    return exportables[property.replace(/([a-z])([A-Z])/g, "$1$2").toLowerCase()];
+  }
+});
```
Comment (rwaldron) [inline lib/easing.js]: I verified these functions by modifying the example.html page that comes with "ease-component" and compared the behavior side-by-side.

Comment (dtex) [reply lib/easing.js]: Cool. Is there an initiative to minimize dependencies generally speaking?

File: lib/animation.js [hunk 2]
```diff
@@ -78,492 +50,499 @@ Animation.TemporalFallback.prototype.stop = function() {
...
-          };
         }
+        source[i] = keyFrame;
 
-        // Replace step values
-        if (typeof keyFrame.step !== "undefined") {
-          keyFrame.value = keyFrame.step === false ?
-            previousVal : previousVal + keyFrame.step;
-        }
+      }, this);
```
Comment (dtex) [inline lib/animation.js]: I think the "this" binding is unnecessary since we're using an arrow function on the opening forEach.

**Comment by dtex** [comment]
This is exciting BTW!

File: lib/easing.js (hunk 1 above)
Comment (rwaldron) [reply lib/easing.js]: Not specifically, I just happened to notice that this one comes from a project that was effectively abandoned. Easing functions aren't novel, so I figured we could just roll our own in there.

**Comment by rwaldron** [comment]
> Were you just knocking out all the extended tests?

Oh, yes, that's correct. Sorry for any confusion there

**Comment by rwaldron** [comment]
appveyor/pr failed because I canceled the build

**Pull request merged**